### PR TITLE
feat(ascend): add weight_quant_matmul for w4a16

### PR DESCRIPTION
**Pull request by yao-fengchen**
feat(ascend): add weight_quant_matmul for w4a16

### CODE REVIEW

File: dlinfer/vendor/ascend/torch_npu_ops.py [hunk 1]
```diff
@@ -341,3 +342,18 @@ def fused_attention(
...
+
+@register_ops(vendor_ops_registry)
+def weight_quant_matmul(
+    x: Tensor,
+    qweight: Tensor,
+    scale: Tensor,
+    offset: Optional[Tensor] = None,
+    bias: Optional[Tensor] = None,
+    group_size: Optional[int] = 0,
+):
```
Comment (jinminxi104) [inline dlinfer/vendor/ascend/torch_npu_ops.py:355]: pls add description of this op

File: dlinfer/ops/llm.py [hunk 2]
```diff
@@ -399,3 +400,23 @@ def get_cache_len(cache: Tensor) -> int:
...
+)
+def weight_quant_matmul(
+    x1: Tensor,
+    x2: Tensor,
+    scale: Tensor,
+    offset: Optional[Tensor],
+    bias: Optional[Tensor],
+    group_size: Optional[int],
+):
+    return vendor_ops_registry["weight_quant_matmul"](
```
Comment (jinminxi104) [inline dlinfer/ops/llm.py:420]: pls add description of this op

File: dlinfer/vendor/ascend/torch_npu_ops.py (hunk 1 above)
Comment (jinminxi104) [inline dlinfer/vendor/ascend/torch_npu_ops.py:355]: pls add description of this op

File: dlinfer/ops/llm.py (hunk 2 above)
Comment (yao-fengchen) [reply dlinfer/ops/llm.py:420]: I will add the description in the next commit.

File: dlinfer/vendor/ascend/torch_npu_ops.py (hunk 1 above)
Comment (yao-fengchen) [reply dlinfer/vendor/ascend/torch_npu_ops.py:355]: I will add the description in the next commit.

**Review by jinminxi104 — APPROVED** [review]

**Pull request merged**